solidity/truffle-examples/random-datasource/README.md

```diff
@@ -0,0 +1,66 @@
+# :wrench: :construction: Testing Oraclize's Random Example.
+
+&nbsp;
+
+This repo is to demonstrate how you would set up an Oraclize smart-contract development environment using Truffle to do most of the heavy lifting for you. Head on over to the `./test` folder to examine the javascript files that thoroughly test the smart-contract, which latter you will find in `./contracts`.
+
+## :page_with_curl:  _Instructions_
+
+Since the random datasource is currently available on the Ethereum mainnet and on all Ethereum public testnets only (Rinkeby, Kovan, Ropsten-revival); and it is *not integrated yet with private blockchains/testrpc/browser-solidity-vmmode*, you will need to **get a mnemonic passphrase and an infura key** to make the example work.
+
+**A)** Go on the infura website to get an infura key: [infura.io](https://infura.io).
+The key is a 32 HEX characters string
+
+**B)** Get a mnemonic passphrase, an easy way is to get the Metamask one:
+[metamask.io](https://metamask.io/)
+
+**C)** Create a new file `apikeys.js` or `.env` and *add the mnemonic passphrase and the infura key to it*, such as:
```

Comment:
Just adding something to a `.env` file doesn't magically make it an environment variable, and so neither the code I wrote in the `truffle.js` nor that you copy/pasted into utils (they should share it, btw, with proper exporting, rather than copy/pasting) would pick up the keys from a `.env` file if the user chooses this option.

Reply:
This line `require('dotenv').config()` adds the `.env` file variables to the environment variable stack.

Reply:
And where is this line? Because as of your latest commit @ time of writing (2ed7a8a) in this PR, neither this line nor the package in the `package.json` exist.

I am aware that that is how you parse `.env` to achieve the environment variable, but I'm pointing out the lack of it in this repo, making the `.env` pointless, and the recommendation for using it in the `README` also pointless.

Reply:
The line is the fix...

Reply:
I am well aware, but you stated the above line as if:

a) It's already in the repo and working
& b) I don't know how to use an environment variable in a node project.

But the point I'm making isn't me asking how to do it, it's that you've suggested a user (who is likely to be a beginner given that they are looking at our truffle examples) to use an environment variable in a `.env` file when that won't work in the repo as it stands at the time of reviewing. 

And so I've raised this entire point in the review precisely because it's _not_ in the repo, and so _doesn't_ work.

Reply:
No, I stated:

this line will now fix the issue. 

If it was not clear the point I'm making with: 

> This line require('dotenv').config() adds the .env file variables to the environment variable stack.

is: there was an issue you pointed out, and I solved it adding that line. I wasn't claiming for obvious reasons that was already in the repo and working.

Reply:
>This line require('dotenv').config() adds the .env file variables to the environment variable stack.

You statement is declarative and conveys no sense of having made a fix whatsoever, nor intention to do so. It simply points out what we all know about how to use an environment variable in a node project.

This review is me pointing out things that need fixing.

Reply:
I am confused. Again, what I want to say here is that editing the file adding `require('dotenv').config()` fix now the issue you pointed out.